--- org/apache/jasper/compiler/Compiler.java	2011-05-24 13:21:43.782077218 +1000
+++ Compiler.java	2011-05-24 13:24:02.252077282 +1000
@@ -737,9 +737,11 @@
      * The default is to use javac from ant.
      */
     private void initJavaCompiler() throws JasperException {
+        boolean disablejsr199 = Boolean.TRUE.toString().equals(
+        System.getProperty("org.apache.jasper.compiler.disablejsr199"));
         Double version = 
             Double.valueOf(System.getProperty("java.specification.version"));
-        if (version >= 1.6 || getClassFor("javax.tools.Tool") != null) {
+        if (!disablejsr199 && (version >= 1.6 || getClassFor("javax.tools.Tool") != null)) {
             // JDK 6 or bundled with jsr199 compiler
             javaCompiler = new Jsr199JavaCompiler();
         } else {
